vhosts.go

```diff
@@ -161,6 +163,8 @@ type VhostSettings struct {
 	Description string `json:"description"`
 	// Virtual host tags
 	Tags VhostTags `json:"tags"`
+	// Type of queue to create in virtual host when unspecified
+	DefaultQueueType string `json:"defaultqueuetype,omitempty"`
```

Comment:
Why is the JSON field not `default_queue_type` here? It is in the management UI's code (the API does not explicitly validate the keys in this case).

Reply:
The inconsistency can be demonstrated using the following test. 

1. Create vhost with `defaultqueuetype`

```
> curl 'https://localhost:5672/api/vhosts/testvhost' -X 'PUT' -u mquser --data-raw '{"name":"testvhost","description":"","tags":"","defaultqueuetype":"quorum"}'
Enter host password for user 'mquser':
```
```
> curl -s 'https://localhost:5672/api/vhosts/testvhost' -u mquser  | jq
Enter host password for user 'mquser':
{
  "name": "testvhost",
  "description": "",
  "tags": [],
  "default_queue_type": "quorum",
  "metadata": {
    "default_queue_type": "quorum",
    "description": "",
    "tags": []
  },
  "tracing": false,
  "cluster_state": {
 ------
  }
}
```

2. Create vhost with `default_queue_type`
```
>curl 'https://localhost:5672/api/vhosts/testvhost1' -X 'PUT' -u mquser --data-raw '{"name":"testvhost1","description":"","tags":"","default_queue_type":"quorum"}'
Enter host password for user 'mquser':
```

```
> curl 'https://localhost:5672/api/vhosts/testvhost1' -u mquser  | jq
Enter host password for user 'mquser':
{
  "name": "testvhost1",
  "description": "",
  "tags": [],
  "default_queue_type": "undefined",
  "metadata": {
    "description": "",
    "tags": []
  },
  "tracing": false,
  "cluster_state": {
------  
  }
}
```

Reply:
Created a [bug](https://github.com/rabbitmq/rabbitmq-server/issues/7734) for rabbitmq-server.

Reply:
Thank you. I have fixed the issue in https://github.com/rabbitmq/rabbitmq-server/pull/7739, let's wait for `3.11.12` to ship.